ilastik/applets/labeling/labelingGui.py

```diff
@@ -752,10 +762,19 @@ def _onLabelRemoved(self, parent, start, end):
         oldcount = self._labelControlUi.labelListModel.rowCount() + 1
         logger.debug("removing label {} out of {}".format( row, oldcount ))
 
-        if self._allowDeleteLastLabelOnly:
+        if self._defTwoInitialLabels and self._allowDeleteLastLabelOnly:
```

Comment:
hmmm, just conceptually (and i am aware that it might have been different before), would it make sense to delete the label first, and then add all this logic? Just in case something goes wrong on label delete, the state will kind of be more consistent? What do you think?

Reply:
Isn't the label already deleted in listModel.py/removeRow() which is called before _onLabelRemoved? In _onLabelRemoved are just the label selection and the colors updated. Don't know what difference it would make if i move it to the end of the function.

Reply:
Okay mabe it makes sense, just to be sure that the whole delete-process was properly executed to that point

Reply:
I will do the same in _addNewLabel